x-pack/legacy/plugins/infra/public/components/source_configuration/log_columns_configuration_form_state.tsx

```diff
@@ -106,6 +106,18 @@ export const useLogColumnsConfigurationFormState = ({
     [formState.logColumns]
   );
 
+  const reorderLogColumns = useCallback(
+    (sourceIndex, destinationIndex) => {
+      const newLogColumns = [...formState.logColumns];
+      newLogColumns.splice(destinationIndex, 0, newLogColumns.splice(sourceIndex, 1)[0]);
```

Comment:
Could this bring the data structure into an inconsistent state if `sourceIndex` and `destinationIndex` were out of bounds? I think it could lead to `undefined`s in the array. It might be worth checking the indices first.